src/asset-platform.js

```diff
@@ -1,11 +1,9 @@
 const { PLATFORM_ARCH } = require("./constants");
 
 const assetPlatform = (fileName) => {
-  if (/.*(mac|darwin|osx).*(-arm).*\.zip/i.test(fileName)) {
-    return PLATFORM_ARCH.DARWIN_ARM64;
-  }
-
-  if (/.*(mac|darwin|osx).*\.zip/i.test(fileName) && !/arm64/.test(fileName)) {
+  if (/.*(mac|darwin|osx).*\.zip$/i.test(fileName)) {
+    if (/universal/.test(fileName)) return PLATFORM_ARCH.DARWIN_UNIVERSAL;
+    if (/arm64/.test(fileName)) return PLATFORM_ARCH.DARWIN_ARM64;
```

Comment:
I think these both need the prefixed `-` to avoid "universal-controller-arm64.app" being detected ahead of "arm64".
The order here should also probably flip for safety

Reply:
Great feedback.  Order switched and added `-` prefix to improve asset matching